packages/dds/map/src/map.ts

```diff
@@ -150,106 +278,220 @@ export class SharedMap extends SharedObject<ISharedMapEvents> implements IShared
         attributes: IChannelAttributes,
     ) {
         super(id, runtime, attributes);
-        this.kernel = new MapKernel(
-            this.serializer,
-            this.handle,
-            (op, localOpMetadata) => this.submitLocalMessage(op, localOpMetadata),
-            () => this.isAttached(),
-            [],
-            this,
-        );
+        this.localValueMaker = new LocalValueMaker(this.serializer);
```

Comment:
there is value in not coupling the core logic of the dds to shared object, this pattern is followed in sequence and merge tree, and it allows easy unit testing of the merge tree, without having to deal with the shared object complexity

Reply:
this file also got huge with these changes

Reply:
I see your point, though we don't test that way today in map.  I can keep them split for this change if you prefer.

Reply:
i don't have a strong opinion, but i do think the pattern has benefits, but it they are not being leveraged it may not really matter.

Reply:
I'll split it back out for this change since it's not needed to scope the exported public API surface, and we can decide if we want to make a change later.